jujugui/static/gui/src/app/store/env/bakery.js

```diff
@@ -438,8 +449,14 @@ YUI.add('juju-env-bakery', function(Y) {
        @return {undefined} Nothing.
        */
       _successfulDischarges: function (originalRequest, jsonMacaroon) {
-        var prefix = this.macaroonName + '=';
-        document.cookie = prefix + btoa(JSON.stringify(jsonMacaroon));
+        if (this.nonCookie) {
+          window.localStorage.setItem(this.macaroonName,
```

Comment:
this localStorage should be passed in when instantiating the module so that you can pass in a stub easy in tests. This means that in app.js you'd pass in `localStorage: window.localStorage`